### ISSUE TITLE
why the default clip_max_norm is 0.1?

### ISSUE THREAD

**Issue by iFighting** [facebookresearch/detr#91]
## ❓ How to do something using DETR

why the default clip_max_norm is 0.1
1. it is so small
2. if i increase the clip_max_norm, can i get the same results with less epochs?

**Comment by alcinos** [comment]
Hi @iFighting 
We have not particularly explored different value of gradient clipping, but 0.1 is fairly standard.

If you increase it, or remove it entirely, you'll see slightly less stable training, and potentially slightly lower final AP, but it is unlikely that you will get faster training (we have never observed it on our side). Feel free to test and report what you get!

I believe I have answered your question and as such I'm closing this. Feel free to reach out if you have further concerns.

**Comment by iFighting** [comment]
i am not meaning training faster, i just mean converge faster with larger gradient, is that possible?

**Comment by alcinos** [comment]
Yes sorry I used imprecise wording. I mean that we didn't observe faster convergence when we remove gradient clipping.
